Improving text understanding on verifying smart contracts page (#7428)
ethereum/ethereum-org-website: src/content/developers/docs/smart-contracts/verifying/index.md

[Smart contracts](/developers/docs/smart-contracts/) are designed to be “trustless”, meaning users shouldn’t have to trust third parties (e.g., developers and companies) before interacting with a contract. As a requisite for trustlessness, users and other developers must be able to verify a smart contract’s source code. Source code verification assures users and developers that the published contract code is the same code running at the contract address on the Ethereum blockchain.

> It is important to make the distinction between "source code verification" and "[formal verification](/developers/docs/smart-contracts/formal-verification/)". Source code verification, which will be explained in detail below, refers to verifying that the given source code of a smart contract in a high-level language (e.g. Solidity) compiles to the same bytecode to be executed at the contract address. On the other hand, formal verification describes verifying the correctness of a smart contract, meaning the contract behaves as expected. Although context-dependent, contract verification usually refers to source code verification.

Reviewer: I agree that this would be improved by further simplification to make it more readable. However, I'm not the biggest fan of using metaphors in technical documentation as they can be unfamiliar to non-Native English speakers. Maybe it would be better to separate this section into three paragraphs?

cc: @emmanuel-awosika any thoughts?

Reviewer 2: > However, I'm not the biggest fan of using metaphors in technical documentation as they can be unfamiliar to non-Native English speakers.

I try to avoid them in writing for this reason. 

> Maybe it would be better to separate this section into three paragraphs? 

This is okay imo.

Author: Ok, let's try it.

Reviewer 2: We could use "while", but that would likely involve joining the two sentences (starting that other sentence with "while" reads a bit awkwardly):

> Source code verification, which will be explained in detail below, refers to verifying that the given source code of a smart contract in a high-level language (e.g. Solidity) compiles to the same bytecode to be executed at the contract address, **while** formal verification describes verifying the correctness of a smart contract (meaning the contract behaves as expected). 

Since that sentence would be really long, maybe we can try "however"?:

> Source code verification, which will be explained in detail below, refers to verifying that the given source code of a smart contract in a high-level language (e.g. Solidity) compiles to the same bytecode to be executed at the contract address. **However**, formal verification describes verifying the correctness of a smart contract, meaning the contract behaves as expected. 

I think the second one achieves what you have in mind w/o complicating things.

Author: I agree to use "however"

> It is important to make the distinction between "source code verification" and "[formal verification](/developers/docs/smart-contracts/formal-verification/)". Source code verification, which will be explained in detail below, refers to verifying that the given source code of a smart contract in a high-level language (e.g. Solidity) compiles to the same bytecode to be executed at the contract address. On the other hand, formal verification describes verifying the correctness of a smart contract, meaning the contract behaves as expected. Although context-dependent, contract verification usually refers to source code verification.

Reviewer 3: ```suggestion
It is important to make the distinction between "source code verification" and "[formal verification](/developers/docs/smart-contracts/formal-verification/)". Source code verification, which will be explained in detail below, refers to verifying that the given source code of a smart contract in a high-level language (e.g. Solidity) compiles to the same bytecode to be executed at the contract address. However, formal verification describes verifying the correctness of a smart contract, meaning the contract behaves as expected. Although context-dependent, contract verification usually refers to source code verification.
```

Pulling in change from discussion in comment thread
